frame/transaction-payment/rpc/runtime-api/src/lib.rs

```diff
@@ -30,5 +32,37 @@ sp_api::decl_runtime_apis! {
 	{
 		fn query_info(uxt: Block::Extrinsic, len: u32) -> RuntimeDispatchInfo<Balance>;
 		fn query_fee_details(uxt: Block::Extrinsic, len: u32) -> FeeDetails<Balance>;
+		fn query_call_info(encoded_call: Bytes) -> Result<RuntimeDispatchInfo<Balance>, DispatchError>;
+		fn query_weight_to_fee(encoded_call: Bytes) -> Result<Balance, DispatchError>;
+		// fn query_call_info(call: GetDispatchInfo, len: u32) -> RuntimeDispatchInfo<Balance>;
+		// fn query_weight_to_fee(call: GetDispatchInfo) -> Balance;
+	}
+}
+
+/// Helper type to implement runtime api functions not included to the pallet
+pub struct TransactionPayment<T, Balance>(PhantomData<(T, Balance)>);
+impl<T, Balance> TransactionPayment<T, Balance>
+where
+	T: pallet_transaction_payment::Config,
+	Balance: Codec + MaybeDisplay,
+{
+	fn query_call_info(encoded_call: Bytes) -> Result<RuntimeDispatchInfo<Balance>, DispatchError> {
+		// let encoded_len = encoded_call.len();
+		// let call: Call = Call::decode(&mut &*encoded_call).map_err(|e| {
+		// 	Error(e) // TODO map decode error to DispatchError::Module to bubble error info
+		// });
+		// Ok (RuntimeDispatchInfo {
+		// 	weight,
+		// 	class,
+		// 	partial_fee: pallet_transaction_payment::Pallet::<T>::query_call_info(len, call) })
+		unimplemented!();
+	}
+
+	fn query_weight_to_fee(encoded_call: Bytes) -> Result<Balance, DispatchError> {
+		// let call: Call = Call::decode(&mut &*encoded_call).map_err(|e| {
+		// 	Error(e) // TODO map decode error to DispatchError::Module to bubble error info
+		// });
+		// Ok(pallet_transaction_payment::Pallet::<T>::query_weight_to_fee(call))
+		unimplemented!();
```

Comment:
Why? You exposed those functions from the pallet. No need to add another indirection. Just call them directly.